packages/next/cli/next-dev.ts

```diff
@@ -73,7 +73,7 @@ const nextDev: cliCommand = (argv) => {
       )
     }
   }
-
+  const inlinePort: number | undefined = args['--port']
```

Comment:
```suggestion
  const allowRetry = !args['--port']
```

This might make more sense to pass to startServer()